api/admin/src/main/java/io/featureprobe/api/auth/LdapAccountValidator.java

```diff
@@ -0,0 +1,106 @@
+import io.featureprobe.api.base.tenant.TenantContext;
+import io.featureprobe.api.dao.entity.Member;
+import io.featureprobe.api.dao.entity.OperationLog;
+import io.featureprobe.api.dao.entity.Organization;
+import io.featureprobe.api.dao.repository.OrganizationRepository;
+import io.featureprobe.api.dto.MemberCreateRequest;
+import io.featureprobe.api.service.MemberService;
+import io.featureprobe.api.service.OperationLogService;
+import io.featureprobe.api.service.OrganizationService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Primary;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.ldap.core.LdapTemplate;
+import org.springframework.ldap.core.support.LdapContextSource;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+import static org.springframework.ldap.query.LdapQueryBuilder.query;
+
+
+@Component(value = "ldap")
+@AllArgsConstructor
+@Qualifier("ldap")
+@Slf4j
+@Primary
+public class LdapAccountValidator implements AccountValidator{
+
+    private MemberService memberService;
+
+    private OperationLogService operationLogService;
+
+
+    private OrganizationRepository organizationRepository;
+
+
+    private LdapTemplate ldapTemplate;
+
+    private LdapContextSource contextSource;
+
+    @Override
+    public Authentication authenticate(Authentication authentication) {
+        UserPasswordAuthenticationToken token = (UserPasswordAuthenticationToken) authentication;
+        try {
+            ldapTemplate.authenticate(query().where("uid").is(token.getAccount()),token.getPassword());
+        }
+         catch (EmptyResultDataAccessException e) {
+             throw new UsernameNotFoundException("Account not find on Ldap Server."+e.getMessage());
+        } catch (AuthenticationCredentialsNotFoundException e) {
+            throw new UsernameNotFoundException("Account password not correct on Ldap Server."+e.getMessage());
+        } catch (Exception e) {
+            log.error("err when authenticate on ldap server {}",e.getMessage());
+        }
+        String memberName = token.getAccount();
+            Optional<Member> memberOptional =  memberService.findByAccount(memberName);
+            OrganizationMemberModel defaultOrganizeMemberModel =
+                    new OrganizationMemberModel(1L, "Default Organize", OrganizationRoleEnum.WRITER);
+            if (!memberOptional.isPresent()) {
```

Comment:
When the LDAP service is unavailable, account creation cannot be performed here.

Reply:
i add a flag check that is wether auth by ldap server.